flang/lib/Lower/OpenMP/OpenMP.cpp

```diff
@@ -2017,37 +2017,46 @@ genTargetOp(lower::AbstractConverter &converter, lower::SymMap &symTable,
     if (dsp.getAllSymbolsToPrivatize().contains(&sym))
       return;
 
+    // Structure component symbols don't have bindings, and can only be
+    // explicitly mapped individually. If a member is captured implicitly
+    // we map the entirety of the derived type when we find its symbol.
+    if (sym.owner().IsDerivedType())
```

Comment:
This change already exists, the previous code was doing the same thing, it's just more explicit in the above case. However, it's likely possible to omit this check + early return as I would imagine it's already being caught by the addition on line 2038. A similar early return is done later, in the body generation when we're handling symbol re-bindings to target arguments. But in either case without these in place, it would be a compiler ICE as opposed to an IR change in cases where derived type members are implicitly captured, as there symbol is the parent and they themselves do not have symbols currently (from my understanding at least).